kafka-ui-react-app/src/components/common/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,130 @@
+import { PER_PAGE } from 'lib/constants';
+import usePagination from 'lib/hooks/usePagination';
+import { range } from 'lodash';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PageControl from './PageControl';
+
+export interface PaginationProps {
+  totalPages: number;
+}
+
+const NEIGHBOURS = 2;
+
+const Pagination: React.FC<PaginationProps> = ({ totalPages }) => {
+  const { page, perPage, pathname } = usePagination();
+
+  const currentPage = page || 1;
+  const currentPerPage = perPage || PER_PAGE;
+
+  const getPath = (newPage: number) =>
+    `${pathname}?page=${Math.max(newPage, 1)}&perPage=${currentPerPage}`;
+
+  const pages = React.useMemo(() => {
```

Comment:
This one seems complicated. Are there any ways to split it?

Reply:
I think we will refactor it in process of dev. Let's say it beta.